tensorrt_llm/_torch/custom_ops/cute_dsl_kernels/blackwell/utils.py

```diff
@@ -0,0 +1,145 @@
+"""
+# Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+
+import ctypes
+from typing import Union
+
+import cutlass._mlir.dialects.cute as _cute_ir
+from cutlass._mlir import ir
+from cutlass.cute.typing import AddressSpace, Numeric, Pointer, Type
+
+
+# WAR for CuTeDSL make_ptr implementation
+class _Pointer(Pointer):
+    """Runtime representation of a pointer that can inter-operate with
+    various data structures, including numpy arrays and device memory.
+    :param pointer: The pointer to the data
+    :type pointer: int or pointer-like object
+    :param dtype: Data type of the elements pointed to
+    :type dtype: Type
+    :param mem_space: Memory space where the pointer resides, defaults generic
+    :type mem_space: _cute_ir.AddressSpace, optional
+    :param assumed_align: Alignment of input pointer in bytes, defaults None
+    :type assumed_align: int, optional
+    :ivar _pointer: The underlying pointer
+    :ivar _dtype: Data type of the elements
+    :ivar _addr_space: Memory space of the pointer
+    :ivar _assumed_align: Alignment of the pointer in bytes
+    :ivar _desc: C-type descriptor for the pointer
+    :ivar _c_pointer: C-compatible pointer representation
+    """
+
+    def __init__(
+        self,
+        pointer,
+        dtype,
+        mem_space: _cute_ir.AddressSpace = _cute_ir.AddressSpace.generic,
+        assumed_align=None,
+    ):
+        self._pointer = pointer
+        self._dtype = dtype
+        self._addr_space = mem_space
+
+        if assumed_align is None:
+            self._assumed_align = dtype.width // 8
+        else:
+            self._assumed_align = assumed_align
+
+        self._desc = None
+        self._c_pointer = None
+        assert int(self._pointer) % self._assumed_align == 0, (
+            f"pointer must be {self._assumed_align} bytes aligned")
+
+    def size_in_bytes(self) -> int:
+        return ctypes.sizeof(ctypes.c_void_p(int(self._pointer)))
```

Comment:
This means the size of pointer object or size of the object pointed by the pointer?

Reply:
the size of pointer object. This code is copied from dsl master as a WAR. After it's released, we could remove it in trtllm and use dsl's function directly.